server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -305,50 +303,74 @@ public void onFailure(Exception e) {
      *
      * @param primaryTerm primary term of translog files.
      * @param files       list of translog files to be deleted.
+     * @param onDeletion runnable to run on deletion of files for both success and failure
      */
-    private void deleteTranslogFilesAsync(long primaryTerm, List<String> files) {
-        transferService.deleteBlobsAsync(
-            ThreadPool.Names.REMOTE_PURGE,
-            remoteBaseTransferPath.add(String.valueOf(primaryTerm)),
-            files,
-            new ActionListener<>() {
-                @Override
-                public void onResponse(Void unused) {
-                    fileTransferTracker.delete(files);
-                    logger.trace("Deleted translogs for primaryTerm={} files={}", primaryTerm, files);
-                }
+    private void deleteTranslogFilesAsync(long primaryTerm, List<String> files, Runnable onDeletion) {
+        try {
+            transferService.deleteBlobsAsync(
+                ThreadPool.Names.REMOTE_PURGE,
+                remoteBaseTransferPath.add(String.valueOf(primaryTerm)),
+                files,
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(Void unused) {
+                        fileTransferTracker.delete(files);
+                        logger.trace("Deleted translogs for primaryTerm={} files={}", primaryTerm, files);
+                        onDeletion.run();
+                    }
 
-                @Override
-                public void onFailure(Exception e) {
-                    logger.error(
-                        () -> new ParameterizedMessage(
-                            "Exception occurred while deleting translog for primaryTerm={} files={}",
-                            primaryTerm,
-                            files
-                        ),
-                        e
-                    );
+                    @Override
+                    public void onFailure(Exception e) {
+                        onDeletion.run();
+                        logger.error(
+                            () -> new ParameterizedMessage(
+                                "Exception occurred while deleting translog for primaryTerm={} files={}",
+                                primaryTerm,
+                                files
+                            ),
+                            e
+                        );
+                    }
                 }
-            }
-        );
+            );
+        } catch (Exception e) {
+            onDeletion.run();
+            throw e;
+        }
     }
 
     /**
      * Deletes metadata files asynchronously using the {@code REMOTE_PURGE} threadpool.
+     * @param metadataFilesToDelete list of metadata files to be deleted.
+     */
+    private void deleteMetadataFilesAsync(List<String> metadataFilesToDelete) {
```

Comment:
nit: Not needed anymore.

Reply:
This is required for cleaning the stale files which happens only once -
https://github.com/opensearch-project/OpenSearch/blob/87bdd0806b77ea4c0ae1eed6830fd1570abd36f6/server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java#L283-L301

The runnable here is no-op-
```
private void deleteMetadataFilesAsync(List<String> metadataFilesToDelete) {
    deleteMetadataFilesAsync(metadataFilesToDelete, () -> {});
}
```